warehouse/legacy/api/xmlrpc/views.py

```diff
@@ -107,18 +108,21 @@ def search(request, spec, operator="and"):
     for field, value in sorted(spec.items()):
         q = None
         for item in value:
+            kw = {"query": item}
+            if field in SEARCH_BOOSTS:
+                kw["boost"] = SEARCH_BOOSTS[field]
             if q is None:
-                q = Q("term", **{field: item})
+                q = Q("match", **{field: kw})
             else:
-                q |= Q("term", **{field: item})
+                q |= Q("match", **{field: kw})
         queries.append(q)
 
     if operator == "and":
         query = request.es.query("bool", must=queries)
     else:
         query = request.es.query("bool", should=queries)
 
-    results = query[:1000].execute()
+    results = query[:100].execute()
```

Comment:
👍

Reply:
This could do with some documentation. Expected results are mysteriously missing from "pip search" if you are searching a common term, and it's not clear why unless digging through the warehouse source code.

Reply:
https://github.com/pypa/warehouse/pull/4281